app/views/loans/show.slim

```diff
@@ -35,13 +35,16 @@
           tr
             th Name
             th Approved
+            th Loaded
             th
 
         tbody
           - @loan.loan_documents.each do |ldoc|
             tr
               td.font-monospace = ldoc.document.file.filename
               td.text-capitalize = ldoc.approved ? 'Yes' : 'No'
+              td =ldoc.file.present? ? 'Yes' : 'No'
               td.text-truncate
                 = link_to 'Download', url_for(ldoc.document.file),
-                          class: 'btn btn-link btn-sm'
+                          class: 'btn btn-link btn-sm', target: '_blank'
+                = link_to "Upload", edit_loan_document_path(ldoc.id)
```

Comment:
@ericksonmc Definitely the best approach would've been thru JS, but also a form screen just for the doc is fine as long as the navigation return to the right places after the action. But please avoid repetition whenever possible.